app/controllers/questionnaire_controller.rb

```diff
@@ -72,33 +74,60 @@ def check_content_hash
   end
 
   def set_response
-    invitation_token = InvitationToken.find_by_token(questionnaire_params[:q])
-    check_invitation_token(invitation_token)
+    the_response = Response.find_by_uuid(questionnaire_params[:uuid])
+    check_response(the_response)
     return if performed?
-    @response = invitation_token.response
+    @response = the_response
     @protocol_subscription = @response.protocol_subscription
     @protocol = @protocol_subscription.protocol
   end
 
+  def redirect_to_next_page
+    first_response = current_user.my_open_responses.first
```

Comment:
Volgens mij kom je altijd op een 404 uit als je een vragenlijst hebt gemist (die je niet mag invullen).

Die my_open_responses daar zit namelijk geen check in om te kijken of een response wel of niet geexpired is. Je doet later in de questionnaire#show route wel een expired check, en als de response dan expired is dan geef je ene 404. Aangezien de eerste open response (not-completed) volgens mij de oudste is, is het volgens mij bijna altijd zo dat als je een meting gemist hebt, dat je dan automatisch uitkomt op een 404.

Zelfs al zou je naar een random open response gaan, dan nog loop je het risico op een 404 terwijl er nog wel andere niet-expired vragenlijsten zijn in te vullen.

Wat je volgens mij wilt doen is de methode Person#my_open_responses aanpassen en daar een response.expired? check inzetten zodat die methode nooit een expired response kan teruggeven. Alle andere code kan dan gewoon zo blijven als het is. En als er geen open responses meer zijn dan komt de student uit op de reward pagina, dus ook als een student alle openstaande vragenlijsten heeft ingevuld kan ze gewoon haar laatste smslinkje nog gebruiken om op haar rewardspagina komen.